src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs

```diff
@@ -12,6 +12,10 @@ public static class GatewayExtensions
         /// Allows the user to control how the gateway behaves.
         /// </summary>
         /// <param name="config">The <see cref="EndpointConfiguration"/> instance to apply the settings to.</param>
+        [ObsoleteEx(
+            TreatAsErrorFromVersion = "4",
+            RemoveInVersion = "5",
+            Message = "Gateway with no configuration is not supported. Use `Gateway(GatewayDeduplicationConfiguration)` instead.")]
```

Comment:
It is still supported in v7. Should this say something like "Provide deduplication configuration."?

Reply:
@mikeminutillo how does this look? Don't know the best way to point to the overload.

```
ReplacementTypeOrMember = "EndpointConfiguration.Gateway(GatewayDeduplicationConfiguration)"
```

Reply:
I think that's probably perfect

Reply:
Great!